modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/Checkpointer.java

```diff
@@ -381,14 +382,13 @@ boolean writePages(
 
         IgniteConcurrentMultiPairQueue<PersistentPageMemory, FullPageId> writePageIds = checkpointDirtyPages.toDirtyPageIdQueue();
 
-        IgniteConcurrentMultiPairQueue<PersistentPageMemory, GroupPartitionId> updatePartitionIds =
-                checkpointDirtyPages.toDirtyPartitionIdQueue();
+        Set<GroupPartitionId> savedPartitionMetas = ConcurrentHashMap.newKeySet();
```

Comment:
So, I believe that this object can be instantiated somewhere inside of checkpoint writer.

Reply:
If each writer has his own set, then we will write the meta several times.

Reply:
Yes, you are right, I missed that part